plat/allwinner/common/plat_helpers.S

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2017-2018, ARM Limited and Contributors. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <arch.h>
+#include <asm_macros.S>
+#include <sunxi_def.h>
+#include <sunxi_mmap.h>
+
+	.globl	plat_crash_console_init
```

Comment:
@antonio-nino-diaz-arm, can you confirm this?

Reply:
That's not true. When the multi console API was introduced we decided to let platform ports do it however they wanted. For example, in Arm platforms we don't intend to use the multi console for crash error messages, as you can see in https://github.com/ARM-software/arm-trusted-firmware/pull/1418. I think that only the MediaTek port uses the multi console driver for crash log output.

The `do_panic` function in common code calls directly the `plat_crash_console_xxx` functions, so you can make them use the multi console or access the UART directly. We decided to write to the UART directly because we didn't want to rely on any memory being correct to print crash logs (you just need the I/O ports to be working). In any case, you still need to implement the `plat_crash_console_xxx` functions, the difference is what you do inside them.

Reply:
Sorry, I was misreading some changes I had locally. What actually happened is that the weak implementations in `plat/common/aarch64/platform_helpers.S` [were updated to use the multi-console API if available](https://github.com/ARM-software/arm-trusted-firmware/commit/17cd67d2af885913427be5c0d8af47647d72896e#diff-fd1d136c860fa038d74d13c395e7fdc2). So we can omit a platform-specific implementation, and fall back to those weak definitions, if we call `console_set_scope`.